src/systems/wheel_slip/WheelSlip.cc

```diff
@@ -0,0 +1,346 @@
+void WheelSlipPrivate::Update(EntityComponentManager &_ecm)
+{
+  for (const auto &linkSurface : this->mapLinkSurfaceParams)
+  {
+    const auto &params = linkSurface.second;
+
+    // get user-defined normal force constant
+    double force = params.wheelNormalForce;
+
+    auto joint = params.joint;
+
+    auto spinAngularVelocityComp =
+        _ecm.Component<components::JointVelocity>(joint);
+
+    if (!spinAngularVelocityComp || spinAngularVelocityComp->Data().empty())
+      continue;
+    double spinAngularVelocity = spinAngularVelocityComp->Data()[0];
+
+    // As discussed in WheelSlipPlugin.hh, the ODE slip1 and slip2
+    // parameters have units of inverse viscous damping:
+    // [linear velocity / force] or [m / s / N].
+    // Since the slip compliance parameters supplied to the plugin
+    // are unitless, they must be scaled by a linear speed and force
+    // magnitude before being passed to ODE.
+    // The force is taken from a user-defined constant that should roughly
+    // match the steady-state normal force at the wheel.
+    // The linear speed is computed dynamically at each time step as
+    // radius * spin angular velocity.
+    // This choice of linear speed corresponds to the denominator of
+    // the slip ratio during acceleration (see equation (1) in
+    // Yoshida, Hamano 2002 DOI 10.1109/ROBOT.2002.1013712
+    // "Motion dynamics of a rover with slip-based traction model").
+    // The acceleration form is more well-behaved numerically at low-speed
+    // and when the vehicle is at rest than the braking form,
+    // so it is used for both slip directions.
+    double speed = params.wheelRadius * std::abs(spinAngularVelocity);
+    double slip1 = speed / force * params.slipComplianceLateral;
+    double slip2 = speed / force * params.slipComplianceLongitudinal;
+
+    math::Vector2d slipCmd;
+    slipCmd.X() = slip1;
+    slipCmd.Y() = slip2;
+
+    components::SlipComplianceCmd newSlipCmdComp(slipCmd);
+
+    auto currSlipCmdComp =
+        _ecm.Component<components::SlipComplianceCmd>(params.collision);
+    if (currSlipCmdComp )
+    {
+      *currSlipCmdComp = newSlipCmdComp;
+    }
+    else
+    {
+      _ecm.CreateComponent(params.collision, newSlipCmdComp);
+    }
+  }
+}
+//////////////////////////////////////////////////
+WheelSlip::WheelSlip()
+  : dataPtr(std::make_unique<WheelSlipPrivate>())
+{
+}
+
+//////////////////////////////////////////////////
+void WheelSlip::Configure(const Entity &_entity,
+    const std::shared_ptr<const sdf::Element> &_sdf,
+    EntityComponentManager &_ecm,
+    EventManager &/*_eventMgr*/)
+{
+  this->dataPtr->model = Model(_entity);
+
+  if (!this->dataPtr->model.Valid(_ecm))
+  {
+    ignerr << "WheelSlip plugin should be attached to a model entity. "
+           << "Failed to initialize." << std::endl;
+    return;
+  }
+
+  auto sdfClone = _sdf->Clone();
+  if (this->dataPtr->Load(_ecm, sdfClone))
+  {
+    this->dataPtr->validConfig = true;
+  }
```

Comment:
How about just:

`this->dataPtr->validConfig = this->dataPtr->Load(_ecm, sdfClone);`

Reply:
a1b6d997612965bd16cdd64dabb09585855e1bdf